### PR TITLE
(PDB-1455 RE-5675) Migrate PDB to ezbake 0.3 & AIO pathing

### PR DESCRIPTION
**Pull request by kbarber**
This patch migrates us to ezbake 0.3 which introduces AIO pathing into the mix.

### CODE REVIEW

**Comment by kbarber** [comment]
This patch is awaiting this one here: https://github.com/puppetlabs/ezbake/pull/250

**Comment by kbarber** [comment]
Cancelled the PR test ^^ it won't pass yet anyway, since it needs the AIO marker.

File: acceptance/setup/pre_suite/90_install_devel_puppetdb.rb [hunk 1]
```diff
@@ -10,6 +10,8 @@
       if (test_config[:database] == :postgres)
         install_postgres(database)
       end
+      # TODO: fix me once merged and ezbake 0.3.8 is released
```
Comment (kbarber) [inline acceptance/setup/pre_suite/90_install_devel_puppetdb.rb]: This needs removal once the other PR over in ezbake is merged/released etc: https://github.com/puppetlabs/ezbake/pull/250

File: project.clj [hunk 2]
```diff
@@ -126,7 +126,7 @@
              :ezbake {:dependencies ^:replace [[puppetlabs/puppetdb ~pdb-version]
                                                [org.clojure/tools.nrepl "0.2.3"]]
                       :name "puppetdb"
-                      :plugins [[puppetlabs/lein-ezbake "0.2.9"
+                      :plugins [[puppetlabs/lein-ezbake "0.3.8-SNAPSHOT"
```
Comment (kbarber) [inline project.clj]: This won't work once merged, we only currently work against a released version, unless we keep the development workflow in place or something. More likely we'll get a release of 0.3.8 soon.

File: puppet/lib/puppet/util/puppetdb/command.rb [hunk 3]
```diff
@@ -32,7 +32,17 @@ def initialize(command, version, certname, payload)
...
+      # We use to_pson still here, to work around the support for shifting
+      # binary data from a catalog to PuppetDB. Attempting to use to_json
+      # we get to_json conversion errors:
+      #
+      #   Puppet source sequence is illegal/malformed utf-8
+      #   json/ext/GeneratorMethods.java:71:in `to_json'
+      #   puppet/util/puppetdb/command.rb:31:in `initialize'
+      #
+      # This is roughly inline with how Puppet serializes for catalogs as of
+      # Puppet 4.1.0. We need a better answer to non-utf8 data end-to-end.
```
Comment (kbarber) [inline puppet/lib/puppet/util/puppetdb/command.rb]: Most definitely the most troublesome part of this patch.

**Comment by kbarber** [comment]
Tests are here for this patch: https://jenkins.puppetlabs.com/view/PuppetDB/view/Other%20Builds/job/platform_puppetdb_intn-sys_source-kbtest/18/

File: puppet/lib/puppet/util/puppetdb/command.rb (hunk 3 above)
Comment (haus) [inline puppet/lib/puppet/util/puppetdb/command.rb]: i didn't even realize puppet 4 still had pson.

Comment (kbarber) [inline puppet/lib/puppet/util/puppetdb/command.rb]: Yeah, seems it does.

**Comment by haus** [comment]
Looks sane to me.

**Comment by shrug** [comment]
I'm :+1: , I released ezbake 0.3.8 earlier, do you want to make the change in this PR or do it in another?

**Comment by kbarber** [comment]
@shrug thanks dude, I've modified this patch, and re-running tests using the shipped ezbake now: https://jenkins.puppetlabs.com/view/PuppetDB/view/Other%20Builds/job/platform_puppetdb_intn-sys_source-kbtest/20/

**Comment by kbarber** [comment]
@pljenkinsro retest this please

**Comment by kbarber** [comment]
That test job I had has worked: https://jenkins.puppetlabs.com/view/PuppetDB/view/Other%20Builds/job/platform_puppetdb_intn-sys_source-kbtest/20/ just testing using the PR testing job now, with BEAKER_TYPE _temporarily_ set to aio.

**Comment by shrug** [comment]
@kbarber I found and fixed a minor issue in ezbake. I'll release 0.3.9 once we're happy. I'm not sure it impacts these test failures (looks like the tests are looking in the wrong place for the embedded db (it's not in /var/lib anymore)

**Comment by kbarber** [comment]
@shrug no, the failures are specifically because the module is still set to use the old path, I'm working on a patch at the moment, I've just been ill this week.

**Comment by kbarber** [comment]
@pljenkinsro retest this please

**Comment by kbarber** [comment]
Completley Green for embedded & postgresql now: https://jenkins.puppetlabs.com/job/platform_puppetdb_intn-sys_source-kbtest/22/.

I've kicked off a PR test with AIO only again, so hopefully we get a green on PR testing also soon.

**Comment by kbarber** [comment]
http://stream1.gifsoup.com/view/554261/the-dance-of-joy-o.gif

**Pull request merged**